products/messaging/frontend/Campaigns/hogflows/testing/hogFlowEditorTestLogic.ts

```diff
@@ -0,0 +1,359 @@
+                        person: {
+                            id: uuid(),
+                            properties: {
+                                email: 'example@posthog.com',
+                            },
+                            name: 'Example person',
+                            url: `${window.location.origin}/person/${uuid()}`,
+                        },
+                        groups: {},
+                        project: {
+                            id: 1,
+                            name: 'Default project',
+                            url: `${window.location.origin}/project/1`,
+                        },
+                        source: {
+                            name: values.campaign.name ?? 'Unnamed',
+                            url: window.location.href.split('#')[0],
+                        },
+                    },
+                    null,
+                    2
+                ),
+                mock_async_functions: true,
+            } as HogflowTestInvocation,
+            {
+                submitTestInvocation: async (testInvocation: HogflowTestInvocation) => {
+                    try {
+                        // TODO: Replace with actual API call when endpoint is available
+                        // const response = await api.hogFlows.test(props.hogFlow.id, {
+                        //     globals: JSON.parse(testInvocation.globals),
+                        //     mock_async_functions: testInvocation.mock_async_functions,
+                        // })
+
+                        // Mock response for now
+                        const response: HogflowTestResult = {
+                            status: 'success',
+                            logs: [
+                                {
+                                    timestamp: new Date().toISOString(),
+                                    level: 'info',
+                                    message: 'Workflow test completed successfully',
+                                },
+                            ],
+                        }
+                        actions.setTestResult(response)
+                        return testInvocation
+                    } catch (error: any) {
+                        console.error('Workflow test error:', error)
+                        lemonToast.error('Error testing workflow')
+                        throw error
+                    }
+                },
+            },
+        ],
+        sampleGlobals: [
+            null as CyclotronJobInvocationGlobals | null,
+            {
+                loadSampleGlobals: async () => {
+                    if (!values.campaign.trigger?.filters) {
+                        return null
+                    }
+
+                    const errorMessage =
+                        'No events match these filters in the last 30 days. Showing an example $pageview event instead.'
+
+                    try {
+                        const query: EventsQuery = {
```

Comment:
I think we should move all this into a shared logic to be used by hog functions and workflows as the majority of it is like for like and would clean things up a bunch